crates/bevy_asset/src/io/embedded/mod.rs

```diff
@@ -324,6 +390,45 @@ mod tests {
         assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
     }
 
+    #[test]
+    fn embedded_asset_path_from_external_crate_root_src_path() {
+        let asset_path = _embedded_asset_path(
+            "my_crate",
+            "/path/to/crate/src".as_ref(),
+            "/path/to/crate/src/foo/plugin.rs".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+        assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
+    }
+
+    // Although extraneous slashes are permitted at the end, e.g., "src////", a
+    // slash at the beginning is not.
+    #[test]
+    #[should_panic(expected = "Failed to find src_prefix \"/src\" in")]
+    fn embedded_asset_path_from_external_crate_bad_root_src_path() {
+        let asset_path = _embedded_asset_path(
+            "my_crate",
+            "/src".as_ref(),
+            "/path/to/crate/src/foo/plugin.rs".as_ref(),
+            "the/asset.png".as_ref(),
+        );
+        assert_eq!(asset_path, Path::new("my_crate/foo/the/asset.png"));
+    }
+
+    // Although extraneous slashes are permitted at the end, e.g., "src////", a
+    // slash at the beginning is not.
+    #[test]
+    #[should_panic(expected = "Failed to find src_prefix \"////src\" in")]
```

Comment:
Ditto RE Windows

Reply:
Let's turn these two similar cases into a comment on one test instead of it being two.